tuf/src/repo_builder.rs

```diff
@@ -210,6 +211,56 @@ where
     _interchange: PhantomData<D>,
 }
 
+impl<'a, D, R> RepoContext<'a, D, R>
+where
+    D: DataInterchange + Sync,
+    R: RepositoryStorage<D>,
+{
+    fn root_keys_changed(&self, root: &Verified<RootMetadata>) -> bool {
+        let root_keys_count = root.root_keys().count();
+        if root_keys_count != self.trusted_root_keys.len() {
+            return true;
+        }
+
+        for &key in &self.trusted_root_keys {
+            if !root.root_keys().any(|x| x == key.public()) {
```

Comment:
this is n^2, how many keys are there usually? if it could be large, maybe collect root keys into a HashSet

Reply:
even though the key count should be pretty small, I switched to a hashset to make it not be a problem in extreme cases.